src/ParseFile.js

```diff
@@ -99,11 +101,15 @@ class ParseFile {
    * @param type {String} Optional Content-Type header to use for the file. If
    *     this is omitted, the content type will be inferred from the name's
    *     extension.
+   * @param metadata {Object} Optional key value pairs to be stored with file object (S3 Only)
+   * @param tags {Object} Optional key value pairs to be stored with file object (S3 Only)
```

Comment:
These are optional and maybe used in other StorageAdapters besides S3.

Can these be retrieved somehow?

Reply:
Add getMetadata() and getTags()?

Reply:
That too, I mean if I fetch a file from database would the metadata and tag be retrieved.

Reply:
no because only the name of the file is actually being stored to the database

Reply:
In S3 case we are retrieving the object from the storage adapter. Doesn't the metadata / tags return with that data? Can you check?

I know that most storage adapters with metadata support only allow setting metadata on creation. In S3 case should `setMetadata` exists? What would happen if you change the metadata and resave?

Reply:
When retrieving object from S3 the metadata is included but the tags are not. Tags would have to be pulled in a separate request. Maybe fetching the metadata and tags could be added in the future? At least with what I have added so far the data can get to S3 so that files can be better managed view other APIs not currently built in Parse.

Reply:
Metadata would be great. I think tags can be added in the future if there is a request for it.

Reply:
How do you suggest I implement this considering the file adapters only return the data? I propose changing the S3 adapter to return an object with the data, and metadata... and updating parse-server to check if an object is coming in, if so, parse out the data and metadata, and if a buffer is coming in ignore metadata?

Reply:
I believe so. We can take a look at it to prevent breaking changes

Reply:
ok... can you look at the PR I have for parse-server too? This goes together with it.